ui/packages/ui/src/app/components/FilterInputFieldComponent.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Flex,
+  FlexItem,
+  FormGroup,
+  Popover,
+  Text,
+  TextInput,
+  TextVariants,
+  ToggleGroup,
+  ToggleGroupItem,
+} from "@patternfly/react-core";
+import {
+  ExclamationCircleIcon,
+  HelpIcon,
+  InfoCircleIcon,
+} from "@patternfly/react-icons";
+import React from "react";
+
+export interface IFilterInputFieldComponentProps {
+  fieldName: string;
+  filterValues: Map<string, string>;
+  setFormData: (formData: Map<string, string>) => void;
+  formData: Map<string, string>;
+  invalidMsg: Map<string, string>;
+  fieldExcludeList: string;
+  fieldIncludeList: string;
+  fieldPlaceholder: string;
+  i18nFilterFieldLabel: string;
+  i18nFilterFieldHelperText: string;
+  i18nInclude: string;
+  i18nExclude: string;
+  i18nFilterFieldInfoMsg: string;
+}
+
+const getInvalidFilterMsg = (
+  filter: string,
+  errorMsg: Map<string, string> | undefined
+) => {
+  let returnVal = "";
+  errorMsg?.forEach((val, key) => {
+    if (key.includes(filter)) {
+      returnVal = val;
+    }
+  });
+  return returnVal;
+};
+
+const getFieldExpression = (
+  data: Map<string, string>,
+  fieldExclude: string,
+  fieldInclude: string
+): string => {
+  return data.get(fieldExclude) || data.get(fieldInclude) || "";
+};
+
+export const FilterInputFieldComponent: React.FunctionComponent<IFilterInputFieldComponentProps> = (
+  props
+) => {
+  const [filterField, setFilterField] = React.useState<string>(
+    getFieldExpression(
+      props.filterValues,
+      props.fieldExcludeList,
+      props.fieldIncludeList
+    )
+  );
+
+  const [fieldSelected, setFieldSelected] = React.useState<string>(
+    props.filterValues.has(props.fieldExcludeList)
+      ? "filedExclude"
```

Comment:
minor misspelling.  should be "fieldExclude" / "fieldInclude" ?

Reply:
done